### ISSUE TITLE
Cognito User Pool: Support ALLOW_* explicit auth flows

### ISSUE THREAD

**Issue by npellegrin** [terraform-providers/terraform-provider-aws#11662]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In the AWS Cognito user pool client resource, there is new explicits auth flows with the `ALLOW_` prefix. Terraform currently only supports the deprecated ones (without prefix) and will block the `ALLOW_` values.

Both versions with and without the `ALLOW_` prefix will work in the AWS API (AWS supports backward compatibility). For me, Terraform shoud not check the values and let the AWS Cognito API trigger an error if invalid.

In the Terraform documentation: https://www.terraform.io/docs/providers/aws/r/cognito_user_pool_client.html
```
explicit_auth_flows - (Optional) List of authentication flows (ADMIN_NO_SRP_AUTH, CUSTOM_AUTH_FLOW_ONLY, USER_PASSWORD_AUTH). 
```

In AWS documentation: https://docs.aws.amazon.com/sdk-for-go/api/service/cognitoidentityprovider/
```
    // The authentication flows that are supported by the user pool clients. Flow
    // names without the ALLOW_ prefix are deprecated in favor of new names with
    // the ALLOW_ prefix. Note that values with ALLOW_ prefix cannot be used along
    // with values without ALLOW_ prefix.
    //
    // Valid values include:
    //
    //    * ALLOW_ADMIN_USER_PASSWORD_AUTH: Enable admin based user password authentication
    //    flow ADMIN_USER_PASSWORD_AUTH. This setting replaces the ADMIN_NO_SRP_AUTH
    //    setting. With this authentication flow, Cognito receives the password
    //    in the request instead of using the SRP (Secure Remote Password protocol)
    //    protocol to verify passwords.
    //
    //    * ALLOW_CUSTOM_AUTH: Enable Lambda trigger based authentication.
    //
    //    * ALLOW_USER_PASSWORD_AUTH: Enable user password-based authentication.
    //    In this flow, Cognito receives the password in the request instead of
    //    using the SRP protocol to verify passwords.
    //
    //    * ALLOW_USER_SRP_AUTH: Enable SRP based authentication.
    //
    //    * ALLOW_REFRESH_TOKEN_AUTH: Enable authflow to refresh tokens.
```

### New or Affected Resource(s)

* `aws_cognito_user_pool_client`

### Potential Terraform Configuration

```hcl
resource "aws_cognito_user_pool_client" "website" {
  name = "website"

  allowed_oauth_flows_user_pool_client = true
  user_pool_id                         = aws_cognito_user_pool.main.id
  generate_secret                      = false
  callback_urls                        = var.website_callback_urls
  logout_urls                          = var.website_logout_urls
  allowed_oauth_flows                  = ["code", "implicit"]
  allowed_oauth_scopes                 = ["email"]
  explicit_auth_flows                  = ["ALLOW_USER_SRP_AUTH", "ALLOW_REFRESH_TOKEN_AUTH"]
}
```

**Comment by woz5999** [comment]
seems to be a duplicate of #10958

**Comment by bflad** [comment]
Hi folks 👋 This does appear to be a duplicate of #10958, which has already been merged and will release with version 2.46.0 of the Terraform AWS Provider, Thursday this week.
